packages/web3-eth/src/utils/watch_transaction_for_confirmations.ts

```diff
@@ -130,27 +130,23 @@ const watchBySubscription = ({
 
 export function watchTransactionForConfirmations<
 	PromiEventEventType extends PromiEventEventTypeBase,
-	ReturnFormat extends ReturnFormatBase,
+	ReturnFormat extends DataFormat,
+	ResolveType = ReceiptInfo,
 >(
 	web3Context: Web3Context<EthExecutionAPI>,
-	transactionPromiEvent: PromiEvent<ReceiptInfo, PromiEventEventType>,
+	transactionPromiEvent: PromiEvent<ResolveType, PromiEventEventType>,
 	transactionReceipt: ReceiptInfo,
 	transactionHash: Bytes,
 	returnFormat: ReturnFormat,
 ) {
-	if (
-		transactionReceipt === undefined ||
-		transactionReceipt === null ||
-		transactionReceipt.blockHash === undefined ||
-		transactionReceipt.blockHash === null
-	)
+	if (!transactionReceipt || !transactionReceipt.blockHash)
```

Comment:
We explicitly check for `=== undefined` and `=== null` elsewhere - I'd just suggest maintaining consistency whether we do it one way or the other